src/serialization/interval_log/mod.rs

```diff
@@ -753,37 +784,26 @@ fn fract_sec_duration(input: &[u8]) -> IResult<&[u8], time::Duration> {
 
             // nanos were invalid utf8. We don't expose these errors, so don't bother defining a
             // custom error type.
-            Err(Err::Error(error_position!(input, ErrorKind::Custom(0))))
+            Err(Err::Error(error_position!(input, ErrorKind::Alpha)))
         }
         Err(e) => Err(e),
     }
 }
 
-// alternate versions of take_while1, used until ergonomic issues with COmpleteByteSlice are resolved
-macro_rules! take_while1_complete (
-  ($input:expr, $submac:ident!( $($args:tt)* )) => ({
-    use nom::ErrorKind;
-    use nom::InputTakeAtPosition;
-
-    let input = $input;
-    match input.split_at_position1(|c| !$submac!(c, $($args)*), ErrorKind::TakeWhile1) {
-      Err(Err::Incomplete(_)) => Ok((&input[input.len()..], input)),
-      res => res,
-    }
-  });
-  ($input:expr, $f:expr) => (
-    take_while1_complete!($input, call!($f));
-  );
-);
-
-named!(fract_sec_tuple<&[u8], (u64, &str)>,
-    do_parse!(
-        secs: flat_map!(recognize!(take_until!(".")), parse_to!(u64)) >>
-        tag!(".") >>
-        nanos_str: map_res!(take_while1_complete!(is_digit), str::from_utf8) >>
-        (secs, nanos_str)
-    )
-);
+fn fract_sec_tuple(input: &[u8]) -> IResult<&[u8], (u64, &str), (&[u8], nom::error::ErrorKind)> {
+    map(
+        tuple((
+            map_res(
+                map_res(recognize(take_until(".")), str::from_utf8),
+                u64::from_str,
+            ),
+            recognize(take_until(".")),
```

Comment:
Not sure I understand why this is here?

Reply:
I am not as well :)

Reply:
@marshallpierce Do you remember? :p

Reply:
Oof, wish I did. There was some quirk of formatting that it was dealing with, like `4.` with the trailing dot or something like that.

Reply:
I think at the time I was really struggling with nom macros so _everything_ seemed confusing and I didn't think to comment that part :'(